components/permissions_gates/channel_permission_gate/channel_permission_gate.jsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class ChannelPermissionGate extends React.PureComponent {
+    static defaultProps = {
+        invert: false
+    }
+
+    static propTypes = {
+
+        /**
+         * Channel to check the permission
+         */
+        channelId: PropTypes.string,
+
+        /**
+         * Team to check the permission
+         */
+        teamId: PropTypes.string,
+
+        /**
+         * Permissions enough to pass the gate (space separated)
```

Comment:
I wrote it as a space separated because avoid PureComponent re-render, but actually have more sense to overwrite the shouldComponentUpdate method to check the perm prop deeply.

Agree the comment could be clearer.

Reply:
And, It is an OR, and the logic behind is that you can build an AND nesting this component, but there is no way to build an "OR" with this. Other option is to use some kind of "operator" property.